# generated by patch-package 6.4.10
#
# command:
#   npx patch-package mysqldump
#
# declared package:
#   mysqldump: 3.2.0
#
diff --git a/node_modules/mysqldump/dist/cjs.js b/node_modules/mysqldump/dist/cjs.js
index 9fb44be..a332e03 100644
--- a/node_modules/mysqldump/dist/cjs.js
+++ b/node_modules/mysqldump/dist/cjs.js
@@ -685,7 +685,9 @@ function compressFile(filename) {
     fs.renameSync(filename, tempFilename);
     const deleteFile = (file) => {
         try {
-            fs.unlinkSync(file);
+            if (fs.existsSync(file)) {
+                fs.unlinkSync(file);
+            }
         }
         catch (_err) {
             /* istanbul ignore next */
diff --git a/node_modules/mysqldump/dist/es.js b/node_modules/mysqldump/dist/es.js
index 5fbd605..9b379a5 100644
--- a/node_modules/mysqldump/dist/es.js
+++ b/node_modules/mysqldump/dist/es.js
@@ -1,10 +1,11 @@
-import { appendFileSync, createReadStream, createWriteStream, renameSync, unlinkSync, writeFileSync } from 'fs';
+import { appendFileSync, createReadStream, createWriteStream, fstat, renameSync, unlinkSync, writeFileSync } from 'fs';
 import { all } from 'deepmerge';
 import { format } from 'sql-formatter';
 import { createConnection } from 'mysql2';
 import { escape } from 'sqlstring';
 import { createGzip } from 'zlib';
 import { createConnection as createConnection$1 } from 'mysql2/promise';
+import fs from 'fs';
 
 /*! *****************************************************************************
 Copyright (c) Microsoft Corporation. All rights reserved.
@@ -683,7 +684,9 @@ function compressFile(filename) {
     renameSync(filename, tempFilename);
     const deleteFile = (file) => {
         try {
-            unlinkSync(file);
+            if (fs.existsSync(file)) {
+                unlinkSync(file);
+            }
         }
         catch (_err) {
             /* istanbul ignore next */
